relengapi/blueprints/slaveloan/tasks.py

```diff
@@ -0,0 +1,199 @@
+        self.retry(exc=exc)
+    # pylint silence
+    # available_slaves = filter(slave_mappings.slave_filter(loan_class), all_slaves)
+    available_slaves = [slave for slave in all_slaves
+                        if slave_mappings.slave_filter(loan_class)(slave)]
+    chosen = random.choice(available_slaves)
+    logger.debug("Chosen Slave = %s" % chosen)
+    return chosen['name']
+
+
+@task(bind=True, max_retries=None)
+@add_to_history(
+    before="Identifying FQDN and IP of {args[1]}",
+    after="Aquired FQDN and IP")
+def fixup_machine(self, machine, loanid):
+    try:
+        fqdn = socket.getfqdn("%s.build.mozilla.org" % machine)
+        ipaddr = socket.gethostbyname("%s.build.mozilla.org" % machine)
+        session = current_app.db.session('relengapi')
+        m = Machines.as_unique(session,
+                               fqdn=fqdn,
+                               ipaddr=ipaddr)
+        l = session.query(Loans).get(loanid)
+        l.machine = m
+        session.commit()
+        l = session.query(Loans).get(loanid)
+    except Exception as exc:  # pylint: disable=W0703
+        logger.exception(exc)
+        self.retry(exc=exc)
+
+
+@task(bind=True)
+@add_to_history(
+    before="Setup tracking bug for {args[1]}",
+    after="Tracking bug {retval!s} linked with loan")
+def bmo_set_tracking_bug(self, machine, loanid):
+    try:
+        session = current_app.db.session('relengapi')
+        l = session.query(Loans).get(loanid)
+        assert l.bug_id
+
+        bug_comment = "Being loaned to %s in Bug %s" % (l.human.ldap, l.bug_id)
+
+        tracking_bug = bugzilla.ProblemTrackingBug(machine, loadInfo=False)
+        try:
+            tracking_bug.refresh()
+        except bzrest.errors.BugNotFound:
+            logger.info("Couldn't find bug, creating it...")
+            tracking_bug.create(comment=bug_comment, depends_on=l.bug_id)
+
+        if tracking_bug.data:
+            data = {
+                "depends_on": {
+                    "add": [l.bug_id],
+                },
+            }
+            if not tracking_bug.data["is_open"]:
+                data["status"] = "REOPENED"
+            tracking_bug.add_comment(bug_comment, data=data)
+        if not tracking_bug.id:
+            raise ValueError("Unexpected result from bmo, retry")
+        return tracking_bug.id
+    except Exception as exc:
+        logger.exception(exc)
+        self.retry(exc=exc)
+
+
+@task(bind=True, max_retries=None)
+@add_to_history(
+    before="Calling slaveapi's disable method",
+    after="Disable request sent")
+def start_disable_slave(self, machine, loanid):
+    try:
+        url = furl(current_app.config.get("SLAVEAPI_URL", None))
+        # XXX: ToDo raise fatal if no slavealloc
+        url.path.add(machine).add("actions").add("disable")
+        postdata = dict(reason="Being loaned on slaveloan %s" % loanid)
+        retry(requests.post, args=(str(url),), kwargs=dict(data=postdata)).json()
```

Comment:
imo yes, the retry provided by redo is protecting against intermittent POST failures, and is short-cycle, then the celery retry mechanic kicks off its timer if this fails over the whole course of those retries.
:white_check_mark: